### DISCUSSION TITLE
How to boot Jetson Xavier NX devkit from NVME?

### DISCUSSION THREAD

**Discussion by NavSriv** [OE4T/meta-tegra#1412] [Q&A] [answered]
I have Jetson Xavier NX Devkit, which boots from SD card. I am using kirkstone branch of meta-tegra to build poky and generate images for "jetson-xavier-nx-devkit" to boot from SD card. 

I have 500GB Samsung 970 Evo Plus NVMe SSD installed on the board and I would like to boot from NVMe. I referred to [Native Boot from SSD on Jetson Xavier (AGX and NX) - No SD Card Needed!](https://www.youtube.com/watch?v=drBKVbOmPEk) but I want to load my own custom built poky image to NVMe instead nvidia provided images. Is there any meta-layer to generate nvme image and documentation to do so? I am successful in copying rootfs to nvme but I am not sure how to remove all dependencies from SD card?

TIA

**Comment by dwalkes** [comment, marked as answer]
I think the instructions you are looking for are at https://github.com/OE4T/meta-tegra/wiki/initrd-flashing-support

**Reply by NavSriv** [reply]
Thanks for your answer. By following this tutorial, I was able to build and flash the images on the board by executing "sudo ./initrd-flash". But once I powered on the board after removing jumper and OTG cable, if I don't have SD card on the board, board doesn't boot up. Is my understanding correct that after doing this whole process -

1. Bootloader, Kernel and Root filesystem should reside on SSD
2. SD card can be removed

Appreciate your time. Thanks

**Reply by madisongh** [reply]
Did you set the `TNSPEC_BOOTDEV:jetson-xavier-nx-devkit` variable to configure the build to boot off the NVME drive?  The example setting [in this section](https://github.com/OE4T/meta-tegra/wiki/initrd-flashing-support#build-configuration) is for a different target MACHINE.

**Reply by NavSriv** [reply]
I had configured the below in conf file -
TNSPEC_BOOTDEV:jetson-xavier-nx-devkit-emmc = "nvme0n1p1"

**Reply by dwalkes** [reply]
Have you tried to reproduce with https://github.com/OE4T/tegra-demo-distro/ ?  If so please let me know which branch and I can attempt to reproduce.  I've booted this configuration successfully from NVMe recently on several branches there I believe.

**Reply by NavSriv** [reply]
It's working for me.. I had replied above. Please check the first comment. The link shared by you helped me to boot the board from NVME

**Reply by dwalkes** [reply]
Got it, so regarding

> But once I powered on the board after removing jumper and OTG cable, if I don't have SD card on the board, board doesn't boot up.

I think this is related to the note at https://github.com/OE4T/meta-tegra/wiki/initrd-flashing-support#notes

> On Xavier NX dev kits (SDcard-based), you must still have an SDcard installed in the slot even if you are booting off an external drive. The SDcard must not have an esp or APP partition on it. You must manually reformat the SDcard, as the flashing process will not do that for you. For all other Jetsons with internal eMMC storage, the eMMC will be erased as part of the flashing process (and re-partitioned/re-populated for those platforms that store some of the bootloader binaries in the eMMC).

**Reply by NavSriv** [reply]
Ohh I am sorry for creating the confusion.. I had created a thread about the issue and it got resolved.. I forgot to update this discussion...
Below is the reference for the problem faced and solution..

[How to boot Jetson Xavier NX devkit from NVME? #1412 ](https://github.com/OE4T/meta-tegra/discussions/1419#discussioncomment-7620666)
